eclair-core/src/main/scala/fr/acinq/eclair/channel/Channel.scala

```diff
@@ -643,17 +642,16 @@ class Channel(val nodeParams: NodeParams, val wallet: EclairWallet, remoteNodeId
       }
 
     case Event(fulfill: UpdateFulfillHtlc, d: DATA_NORMAL) =>
-      Try(Commitments.receiveFulfill(d.commitments, fulfill)) match {
-        case Success(Right((commitments1, origin, htlc))) =>
+      Commitments.receiveFulfill(d.commitments, fulfill) match {
+        case Success((commitments1, origin, htlc)) =>
           // we forward preimages as soon as possible to the upstream channel because it allows us to pull funds
           relayer ! Relayer.ForwardFulfill(fulfill, origin, htlc)
           stay using d.copy(commitments = commitments1)
-        case Success(Left(_)) => stay
```

Comment:
@t-bast please double-check this thx.

Reply:
This looks fine, the code for `receiveFulfill` never returned a `Left`, so it looks like this was dead code, right?